pcap/slicer.go

```diff
@@ -16,43 +16,53 @@ import (
 // modifying or copying the file.
 type Slicer struct {
 	slices []Slice
+	slice  Slice
 	file   *os.File
-	reader *io.SectionReader
+	eof    bool
 }
 
 func NewSlicer(file *os.File, index *Index, span nano.Span) (*Slicer, error) {
 	slices, err := GenerateSlices(index, span)
 	if err != nil {
 		return nil, err
 	}
-	first := slices[0]
-	return &Slicer{
-		slices: slices[1:],
+	s := &Slicer{
+		slices: slices,
 		file:   file,
-		reader: first.NewReader(file),
-	}, nil
+	}
+	return s, s.next()
+}
+
+func (s *Slicer) next() error {
+	if len(s.slices) == 0 {
+		s.eof = true
+		return nil
+	}
+	s.slice = s.slices[0]
+	s.slices = s.slices[1:]
+	_, err := s.file.Seek(int64(s.slice.Offset), 0)
+	return err
 }
 
 func (s *Slicer) Read(b []byte) (int, error) {
-	for s.reader != nil {
-		n, err := s.reader.Read(b)
-		if n != 0 {
-			if err == io.EOF {
-				err = nil
-			}
-			return n, err
-		}
-		if err != io.EOF {
-			return 0, err
+	if s.eof {
+		return 0, io.EOF
+	}
+	p := b
+	if uint64(len(p)) > s.slice.Length {
+		p = p[:s.slice.Length]
+	}
+	n, err := s.file.Read(p)
+	if n != 0 {
```

Comment:
Per the godocs, file.Read will always return 0, io.EOF. Unless you want to return `io.EOF` you should check for it before this if statement.

Reply:
I think the logic is correct here.  From go docs on io.Reader
```
When Read encounters an error or end-of-file condition after successfully reading n > 0 bytes, it returns the number of bytes read. It may return the (non-nil) error from the same call or return the error (and n == 0) from a subsequent call. 
```